runtime-rs: Add Configurable Compilation for Dragonball in Runtime-rs (#10312)
kata-containers/kata-containers: docs/install/kata-containers-3.0-rust-runtime-installation-guide.md

```
After running the command above, the default config file `configuration.toml` will be installed under `/usr/share/defaults/kata-containers/`,  the binary file `containerd-shim-kata-v2` will be installed under `/usr/local/bin/` .

> ### build Kata 3.0 Rust Runtime Shim without Dragonball VMM

Reviewer: ```suggestion
### Build Shim Without Builtin Dragonball VMM
```

> The Runtime-rs is built with build-in VMM Dragonball by default.
> If you want to build the runtime-rs without the Buildin VMM, you need to run `make` with `DRAGONBALL=no` as follows.

Reviewer: ```suggestion
By default, building `runtime-rs` includes the `Dragonball VMM`. To build without it, use `make DRAGONBALL=no`.
```

```
After running the command above, the default config file `configuration.toml` will be installed under `/usr/share/defaults/kata-containers/`,  the binary file `containerd-shim-kata-v2` will be installed under `/usr/local/bin/` .

### Build Shim Without Builtin Dragonball VMM

By default, `runtime-rs` includes the `Dragonball` VMM. However, when a `non-dragonball` hypervisor is set as the default hypervisor, `dragonball` will not be compiled. For example, if you want to use `qemu` as the default hypervisor,  use `make DEFAULT_HYPERVISOR=qemu`. 

```bash
$ cd kata-containers/src/runtime-rs

> $ make DEFAULT_HYPERVISOR=qemu

Reviewer 2: `DEFAULT_HYPERVISOR` looks a little bit weird for me. What about `HYPERVISOR`?

Author: Currently, there is no variable named HYPERVISOR in the Makefile, and DEFAULT_HYPERVISOR controls the default hypervisor configuration. It appears multiple times in the Makefile. If we introduce HYPERVISOR, does it mean we need to replace all instances of DEFAULT_HYPERVISOR with HYPERVISOR? Otherwise, there will be two variables controlling the hypervisor in use.

Reviewer 2: Yes. I think we have to do that.

Author: The modifications have already been made.
